packages/components/select/src/components/osds-select/osds-select.tsx

```diff
@@ -120,6 +120,8 @@ export class OsdsSelect implements OdsSelectAttribute, OdsSelectEvent, OdsSelect
   async componentDidLoad() {
     this.setSelectOptions();
     await this.updateSelectOptionStates(this.value);
+    this.observer = new MutationObserver(async () => this.selectedOptionLabel = await this.optionSelected?.getLabel() || '');
+    this.observer?.observe(this.controller.selectOptions[0], { childList: true });
```

Comment:
did you check if there is no options?
a possible use case would be a select with asynchronous options, first render is empty, then integrator get the results and set the options accordingly